### PR TITLE
chore: update Go to 1.21

### PR DESCRIPTION
**Pull request by smira**
This fixes a problem in the `RouteSpecController` which is due to a subtle (but correct) change in the behavior in the `stdlib`.

Also some small (but should be safe) bumps.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/siderolabs/talos
 
-go 1.20
+go 1.21
```
Comment (frezbo) [inline go.mod:3]: 1.21.0?

**Review by frezbo — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (smira) [reply go.mod:3]: I'm kinda torn on that, I don't see why we should do a minor version there? unless we depend on something to be exactly 1.21.0, but not 1.21.1 ? I understand this is a compatibility guarantee, or a constraint

File: internal/app/machined/pkg/controllers/network/route_spec.go [hunk 2]
```diff
@@ -119,6 +120,17 @@ func (ctrl *RouteSpecController) Run(ctx context.Context, r controller.Runtime,
 	}
 }
 
+// netipPrefixBitsCorrected returns the number of bits in the prefix, corrected for zero value to have bits of 0.
+//
+// Go stdlib returns -1 for zero value, which is not what we want.
+func netipPrefixBitsCorrected(p netip.Prefix) int {
```
Comment (frezbo) [inline internal/app/machined/pkg/controllers/network/route_spec.go:126]: can we fix our code to be same as expected spec, or am I missing something here?

Comment (smira) [reply internal/app/machined/pkg/controllers/network/route_spec.go:126]: we use it to interface with Linux kernel, and there it expects DstLength to be 0 if the DstAddress is nil.

zero `netip.Prefix` gives us exactly `nil` address, but length (bits in the prefix) became wrong now, as Linux accepts `uint8`, and -1 becomes 255.

File: go.mod (hunk 1 above)
Comment (smira) [reply go.mod:3]: @DmitriyMV please help

Comment (smira) [reply go.mod:3]: @DmitriyMV please help

**Pull request merged**